File: UNIVERCITY_MANAGEMENT SYSTEM/UMI-CLIENT/src/app/services/ecole-doctorale.service.ts
// EcoleDoctorale.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { EcoleDoctorale } from '../models/ecole-doctorale.module';

@Injectable({
  providedIn: 'root'
})
export class EcoleDoctoraleService {
  private apiUrl = 'http://localhost:9016/api/ecole-doctorales';

  constructor(private http: HttpClient) { }

  getAllEcoleDoctorales(): Observable<EcoleDoctorale[]> {
    return this.http.get<EcoleDoctorale[]>(this.apiUrl);
  }
  getEcoleDoctoraleById(id: number): Observable<EcoleDoctorale> {
    return this.http.get<EcoleDoctorale>(`${this.apiUrl}/${id}`);
  }

  createEcoleDoctorale(ecoleDoctorale: EcoleDoctorale): Observable<EcoleDoctorale> {
    return this.http.post<EcoleDoctorale>(this.apiUrl, ecoleDoctorale);
  }

  updateEcoleDoctorale(id: number, ecoleDoctorale: EcoleDoctorale): Observable<EcoleDoctorale> {
    return this.http.put<EcoleDoctorale>(`${this.apiUrl}/${id}`, ecoleDoctorale);
  }

  deleteEcoleDoctorale(id: number): Observable<string> {
    return this.http.delete<string>(`${this.apiUrl}/${id}`);
  }

}

File: UMI-API/ecole-doctorale/src/main/java/com/umi/recherche/ecoledoctorale/service/EcoleDoctoraleService.java
package com.umi.recherche.ecoledoctorale.service;

import com.umi.recherche.ecoledoctorale.dto.EcoleDoctoraleRequestDTO;
import com.umi.recherche.ecoledoctorale.dto.EcoleDoctoraleResponseDTO;
import com.umi.recherche.ecoledoctorale.entity.EcoleDoctorale;

import java.util.List;

public interface EcoleDoctoraleService {
    EcoleDoctoraleResponseDTO createEcoleDoctorale(EcoleDoctoraleResponseDTO ecoleDoctoraleResponseDTO);
    public List<EcoleDoctorale> getAllEcoleDoctorales();
    public EcoleDoctorale getEcoleDoctoraleById(Long id);
    public EcoleDoctoraleResponseDTO updateEcoleDoctorale(Long id, EcoleDoctoraleRequestDTO ecoleDoctoraleRequestDTO);
    public Boolean deleteEcoleDoctorale(Long id);
}
